jax/_src/numpy/util.py

```diff
@@ -265,3 +266,10 @@ def _where(condition: ArrayLike, x: ArrayLike, y: ArrayLike) -> Array:
   except:
     is_always_empty = False  # can fail with dynamic shapes
   return lax.select(condition, x_arr, y_arr) if not is_always_empty else x_arr
+
+
+def broadcastable_shapes(shape):
+  # yields all shapes that can be broadcasted to the given shape
+  # https://numpy.org/doc/stable/user/basics.broadcasting.html#general-broadcasting-rules
+  for i in range(len(shape) + 1):
+    yield from itertools.product(*({dim, 1} for dim in shape[i:]))
```

Comment:
Sorry I missed this – it looks like this is no longer used anywhere, so we should delete it.